test/testdata/infer/singleton_of_singleton.rb.cfg.exp

```diff
@@ -35,22 +35,22 @@ subgraph "cluster_::<Class:A>#foo" {
     "bb::<Class:A>#foo_1" -> "bb::<Class:A>#foo_1" [style="bold"];
 }
 
-subgraph "cluster_::<Class:<Class:A>>#bar" {
-    label = "::<Class:<Class:A>>#bar";
+subgraph "cluster_::<Class:A>#bar" {
+    label = "::<Class:A>#bar";
     color = blue;
-    "bb::<Class:<Class:A>>#bar_0" [shape = invhouse];
-    "bb::<Class:<Class:A>>#bar_1" [shape = parallelogram];
+    "bb::<Class:A>#bar_0" [shape = invhouse];
+    "bb::<Class:A>#bar_1" [shape = parallelogram];
 
-    "bb::<Class:<Class:A>>#bar_0" [
-        label = "block[id=0]()\l<self>: T.class_of(T.class_of(A)) = cast(<self>: NilClass, AppliedType {\l  klass = <S <S <C <U A>> $1> $1>\l  targs = [\l    <C <U <AttachedClass>>> = SelfTypeParam(<S <S <C <U A>> $1> $1><C <U <AttachedClass>>>)\l  ]\l});\l<returnMethodTemp>$2: T.class_of(T.class_of(A)) = <self>\l<finalReturn>: T.noreturn = return <returnMethodTemp>$2: T.class_of(T.class_of(A))\l<unconditional>\l"
```

Comment:
the type here has changed substantially.
This seems like a new bug.

Reply:
This also is a fix and not a bug. The source file in question here is
```ruby
class A
  def self.foo
    def self.bar
      self
    end
  end
end
```
We previously treated `bar` as though it were defined on `T.class_of(T.class_of(A))`, which is not correct (c.f. issue #122). The new type shown here is what it should have been.